stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/RocksCheckpointer.java

```diff
@@ -103,32 +79,38 @@ private static void cleanupLocalCheckpoints(File checkpointsDir, String checkpoi
         }
     }
 
-    private static Pair<String, CheckpointMetadata> getLatestCheckpoint(
-        String dbPrefix, CheckpointStore checkpointStore) throws IOException {
+    private static CheckpointInfo getLatestCheckpoint(
+        String dbPrefix, CheckpointStore checkpointStore) {
+        List<CheckpointInfo> checkpoints = getCheckpoints(dbPrefix, checkpointStore);
+        return checkpoints.get(0);
```

Comment:
can we deal with an empty list ?
maybe it should not happen, but in case it happen we will have a bad runtime exception without a meaningful error message

Reply:
Ideally this will never happen as there should always be a `nullCheckpoint` added to the list. But we add a explicit check to ensure proper error message